connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -761,6 +803,44 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
 
     }
 
+    @SuppressWarnings("unchecked")
+    RestartRequest recordToRestartRequest(ConsumerRecord<String, byte[]> record, SchemaAndValue value) {
+        String connectorName = record.key().substring(RESTART_PREFIX.length());
+        if (value.value() == null) {
+            log.error("Ignoring restart request because it is unexpectedly null");
+            return null;
+        }
+        if (!(value.value() instanceof Map)) {
+            log.error("Ignoring restart request because the value is not a Map but is {}", value.value().getClass());
+            return null;
+        }
+
+        Map<String, Object> valueAsMap = (Map<String, Object>) value.value();
+
+        Object failed = valueAsMap.get(ONLY_FAILED_FIELD_NAME);
+        if (failed == null) {
+            log.warn("Invalid data for restart request '{}' field was missing, defaulting to {}", ONLY_FAILED_FIELD_NAME, ONLY_FAILED_DEFAULT);
+            failed = ONLY_FAILED_DEFAULT;
+        }
+        if (!(failed instanceof Boolean)) {
+            log.warn("Invalid data for restart request '{}' field should be a Boolean but is {}, defaulting to {}", ONLY_FAILED_FIELD_NAME, failed.getClass(), ONLY_FAILED_DEFAULT);
+            failed = ONLY_FAILED_DEFAULT;
+        }
+        boolean onlyFailed = (Boolean) failed;
```

Comment:
Following up on the comment above, here's how we could write this and avoid autoboxing/unboxing while using the more readable primitives. 
```suggestion
        if (!(failed instanceof Boolean)) {
            log.warn("Invalid data for restart request '{}' field should be a Boolean but is {}, defaulting to {}",
                    ONLY_FAILED_FIELD_NAME, failed == null ? "null" : failed.getClass(), ONLY_FAILED_DEFAULT);
            onlyFailed = ONLY_FAILED_DEFAULT;
        } else {
            onlyFailed = (Boolean) failed;
        }
```

Reply:
I Agree and I fixed it like you suggested.